azext_iot/digitaltwins/providers/connection/builders.py

```diff
@@ -0,0 +1,285 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import json
+from azext_iot.common.embedded_cli import EmbeddedCLI
+from azext_iot.sdk.digitaltwins.controlplane.models import AzureDataExplorerConnectionProperties
+from abc import ABC
+from knack.util import CLIError
+from knack.log import get_logger
+from knack.prompting import prompt_y_n
+from azext_iot.digitaltwins.common import (
+    DT_IDENTITY_ERROR,
+    ERROR_PREFIX,
+    FINISHED_CHECK_RESOURCE_LOG_MSG,
+    ADX_ROLE_MSG,
+    RBAC_ROLE_MSG,
+    TRY_ADD_ROLE_LOG_MSG,
+    PRESENT_ADD_ROLE_LOG_MSG,
+    FINISHED_ADD_ROLE_LOG_MSG,
+    SKIP_MSG,
+    FAIL_RBAC_MSG,
+    FAIL_GENERIC_MSG,
+    ABORT_MSG,
+    ADD_INPUT_MSG,
+    CONT_INPUT_MSG
+)
+
+logger = get_logger(__name__)
+
+
+class AdxConnectionValidator(ABC):
```

Comment:
This is a full concrete class implementation, what is the purpose of deriving from ABC? This can be a vanilla class.